request-deferrer/index.js

```diff
@@ -0,0 +1,158 @@
+var ENDPOINT = 'api/quotations';
+
+// Register the worker and show the list of quotations.
+if (navigator.serviceWorker.controller) {
+  loadQuotations();
+} else {
+  navigator.serviceWorker.oncontrollerchange = function() {
+    this.controller.onstatechange = function() {
+      if (this.state === 'activated') {
+        loadQuotations();
+      }
+    };
+  };
+  navigator.serviceWorker.register('service-worker.js');
+}
+
+// This forces a sync with the server. The POST request will never
+// reach the network. It is intended to be intercepted by the
+// Service Worker forcing it to do a sync with the actual remote
+// server.
+window.addEventListener('online', function() {
```

Comment:
On Fedora, the `online` event is triggered when you enable WiFi, even before connecting (so in general it could be triggered even if you can't connect to any network).